File: src/main/java/com/amazonClone/amazon/service/CartService.java
package com.amazonClone.amazon.service;

import com.amazonClone.amazon.dto.request.AddToCartRequest;
import com.amazonClone.amazon.dto.request.UpdateCartItemRequest;
import com.amazonClone.amazon.dto.response.CartResponse;
import com.amazonClone.amazon.entity.User;

public interface CartService {

    CartResponse getCart(User user);

    CartResponse addToCart(User user, AddToCartRequest request);

    CartResponse updateCartItem(User user, Long cartItemId, UpdateCartItemRequest request);

    CartResponse removeCartItem(User user, Long cartItemId);

    CartResponse clearCart(User user);
}

File: amazon-clone-frontend/src/services/cartService.js
import axiosInstance from "../api/axiosInstance";


const cartService = {
    getCart: async () => {
        const { data } = await axiosInstance.get("/cart");
        return data.data; // CartResponse
    },

    addToCart: async ({ productId, quantity }) => {
        const { data } = await axiosInstance.post("/cart/add", { productId, quantity });
        return data.data;
    },

    updateCartItem: async (cartItemId, { quantity }) => {
        const { data } = await axiosInstance.put(`/cart/items/${cartItemId}`, { quantity });
        return data.data;
    },

    removeCartItem: async (cartItemId) => {
        const { data } = await axiosInstance.delete(`/cart/items/${cartItemId}`);
        return data.data;
    },

    clearCart: async () => {
        const { data } = await axiosInstance.delete("/cart/clear");
        return data.data;
    },
};

export default cartService;